WordPress/src/test/java/org/wordpress/android/ui/stories/SaveStoryGutenbergBlockUseCaseTest.kt

```diff
@@ -0,0 +1,288 @@
+        // Given
+        val mediaFiles: ArrayList<MediaFile> = setupFluxCMediaFiles(emptyList = true)
+        editPostRepository.set { PostModel() }
+
+        // When
+        saveStoryGutenbergBlockUseCase.buildJetpackStoryBlockInPost(
+                editPostRepository,
+                mediaFiles
+        )
+
+        // Then
+        Assertions.assertThat(editPostRepository.content).isEqualTo(blockWithEmptyMediaFiles)
+    }
+
+    @Test
+    fun `post with non-empty Story block is set given a non-empty mediaFiles array`() {
+        // Given
+        val mediaFiles: ArrayList<MediaFile> = setupFluxCMediaFiles(emptyList = false)
+        editPostRepository.set { PostModel() }
+
+        // When
+        saveStoryGutenbergBlockUseCase.buildJetpackStoryBlockInPost(
+                editPostRepository,
+                mediaFiles
+        )
+
+        // Then
+        Assertions.assertThat(editPostRepository.content).isEqualTo(blockWithNonEmptyMediaFiles)
+    }
+
+    @Test
+    fun `builds non-empty story block string from non-empty mediaFiles array`() {
+        // Given
+        val mediaFileDataList: ArrayList<StoryMediaFileData> = setupMediaFileDataList(emptyList = false)
+
+        // When
+        val result = saveStoryGutenbergBlockUseCase.buildJetpackStoryBlockStringFromStoryMediaFileData(
+                mediaFileDataList
+        )
+
+        // Then
+        Assertions.assertThat(result).isEqualTo(blockWithNonEmptyMediaFiles)
+    }
+
+    @Test
+    fun `builds empty story block string from empty mediaFiles array`() {
+        // Given
+        val mediaFileDataList: ArrayList<StoryMediaFileData> = setupMediaFileDataList(emptyList = true)
+
+        // When
+        val result = saveStoryGutenbergBlockUseCase.buildJetpackStoryBlockStringFromStoryMediaFileData(
+                mediaFileDataList
+        )
+
+        // Then
+        Assertions.assertThat(result).isEqualTo(blockWithEmptyMediaFiles)
+    }
+
+    @Test
+    fun `obtain a mediaFileData from a MediaFile with a temporary id`() {
+        // Given
+        val mediaFile = getOneMediaFile(1)
+
+        // When
+        val mediaFileData = saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
+                mediaFile,
+                TEMPORARY_ID_PREFIX + 1
+        )
+
+        // Then
+        Assertions.assertThat(mediaFileData.alt).isEqualTo("")
+        Assertions.assertThat(mediaFileData.id).isEqualTo(TEMPORARY_ID_PREFIX + 1)
+        Assertions.assertThat(mediaFileData.link).isEqualTo(
+                "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+        )
+        Assertions.assertThat(mediaFileData.type).isEqualTo("image")
+        Assertions.assertThat(mediaFileData.mime).isEqualTo(mediaFile.mimeType)
+        Assertions.assertThat(mediaFileData.caption).isEqualTo("")
+        Assertions.assertThat(mediaFileData.url).isEqualTo(mediaFile.fileURL)
```

Comment:
I'm not a big fan of multiple Assertions within one test method 😅 To note, this is learnt behavior from Jirka 😆 so yes, he taught me that a unit test should have one single reason to fail, hence the name `unit` 😉 

So the assertion I am expecting here based on the test name is that `mediaFileData` is not null and was returned from the use case. All the other assertions would be separate tests as they are testing different things.

Reply:
> he taught me that a unit test should have one single reason to fail, hence the name unit

I see, this is understandable 👍 

> is that mediaFileData is not null and was returned from the use case

This got me thinking - `buildMediaFileDataWithTemporaryId` returns a new object and cannot return null. Hence, it doesn't make sense to write a test to check the result is not null. It will always be not null, right? That's why I'm checking for the properties of the object as I know the resulting object should look like `This` given the source object looks like `That` , does that make sense? Wondering how we could write things here. Thoughts?

Reply:
Indeed. I totally understand this. Normally I would just duplicate the method logic meaning the parts under the given and when and then verify each property of the object but honestly, in this case, we can probably just leave this as is to not cause too much duplicate logic. So yes, it's up to you :)

Reply:
Thanks for the explanation! I think it makes sense to verify that the properties of the source object have been mapped to the target object in the way defined by each of the assertion statements. If any of these fail, I want the test to fail 👍

Reply:
Indeed. Makes sense 🙇 Could we make the name of the test, something like `verify that all properties of mediaFileData that are created from buildMediaFileDataWithTemporaryId are correct`? I want it to be a bit a clearer that these assertions are intention.

Reply:
yes, renamed in 10a8ee0562 - also dropped the second word `that` because of line length reasons